### ISSUE TITLE
Error when computing scores, `IndexError: index 0 is out of bounds for axis 0 with size 0`

### ISSUE THREAD

**Issue by RicherMans** [audioanalytic/psds_eval#4]
An error is thrown with PSDS 0.3.0 and above, if the ground truth dataframe contains more than the expected four columns `filename`, `event_label`,`onset`, `offset`.

The code to reproduce this issue is as follows:

```python
gt = pd.DataFrame(
    {
        'filename': 'a.wav',
        'onset': 0.,
        'offset': 4.,
        'event_label': 'fake',
        'whoops': 1
    },
    index=[0])
durations = pd.DataFrame(
    {
        'filename': 'a.wav',
        'duration': 10.
    },
    index=[0])

psds = psds_eval.PSDSEval(
    ground_truth=gt,
    metadata=durations,
    dtc_threshold=0.5,
    gtc_threshold=0.5,
    cttc_threshold=0.5,
)

detections = pd.DataFrame(
    {
        'filename': 'a.wav',
        'onset': 0.,
        'offset': 2.,
        'event_label': 'fake'
    },
    index=[0])
thres_f1, _ = psds.compute_macro_f_score(detections)
print(f"F1 is {thres_f1}")
```
yields `IndexError: index 0 is out of bounds for axis 0 with size 0`.
For more information refer to #2 

A simple temporaray fix would be to just pass the necessary columns to psds:
```python
psds = psds_eval.PSDSEval(
    ground_truth=gt[["filename", "onset", "offset", "event_label"]],
    metadata=durations,
    dtc_threshold=0.5,
    gtc_threshold=0.5,
    cttc_threshold=0.5,
)
```

**Comment by cbilen** [comment]
Just to make it more clear for other readers, this question was answered in another thread with the temporary workaround as given above. We will add a fix in the next version, but until then the workaround above should solve the problems for other users who may encounter a similar issue.

**Comment by tgrainge** [comment]
@RicherMans I am taking a look at fixing this so that the work around [mentioned here](https://github.com/audioanalytic/psds_eval/issues/2#issuecomment-842326737) is no longer required.

**Comment by tgrainge** [comment]
Hi @RicherMans, 

The commit bf57fc349ae688c30db68b6f2e0f423e1b5cc850 is now available in the release 0.5.0. This will solve the issue you have raised to us. psds will now only take the columns it requires even if extras are provided. 

Kind Regards 

Tom
